recursion.rb

```diff
@@ -1,16 +1,58 @@
 
+#RC: n! = n * (n-1)!
+#BC: 1! = 1
+#BC: 0! = 1
+
+
 def fact(n)
+  if n == 1 || n == 0
+    return 1 #cant return n here bc of the second BC
+  end
+  return n * fact(n-1)
 end
 
+#BC: fib(1) = 1
+#BC: fib(2) = 1
+#fib(3) = fib(2) + fib(1)
+#fib(4) = fib(3) + fib(2)
+#fib(n) = fib(n-1) + fib(n-2)
+
 def fib(n)
+  return 0 if n == 0
+  return 1 if n == 1
+
+  fib(n - 1) + fib(n - 2)
+  
+  end
+
 end
 
-def pal(s)
+def pal(word)
+  return true if word.length <= 1
+  return false if word[0] != word[-1]
+
+  word.slice!(0)
+  word.slice!(-1)
+
+  pal(word)
 end
 
+
 def binary(n)
+  binary_helper(n,"","")
+end
+
+def binary_helper(digits, binary, result)
+  if digits == 0
+    result += binary + " "
+  else
+  result = binary_helper(digits - 1, binary +"0", result)
+  result = binary_helper(digits - 1, binary +"1",result)
+  end
+  return result
 end
```

Comment:
Thank you ^_^

Sent from my iPhone

> On May 15, 2016, at 9:46 PM, malderi <notifications@github.com> wrote:
> 
> In recursion.rb:
> 
> >  def binary(n)
> > +  binary_helper(n,"","")
> > +end
> > +
> > +def binary_helper(digits, binary, result)
> > +  if digits == 0
> > +    result += binary + " "
> > +  else
> > +  result = binary_helper(digits - 1, binary +"0", result)
> > +  result = binary_helper(digits - 1, binary +"1",result)
> > +  end
> > +  return result
> >  end
> Good solution.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>